### ISSUE TITLE
False negative for RANGE_ARRAY_INDEX

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#2029]
Hi, I found a false negative about the rule RANGE_ARRAY_INDEX. Please read the example below:

```java
int getLiteral() {
  return 4;
}
public int getValue() {
    return array[getLiteral()];  // should report a warning here
}
```
Based on my experience, SpotBugs is a powerful tool to analyze the literal return value. Hence, I think this is a false negative. Thanks for your consideration.

SpotBugs Version: Latest

**Comment by Zustin** [comment]
I found rules DMI_EMPTY_DB_PASSWORD and DMI_CONSTANT_DB_PASSWORD have similar issues.

**Comment by Zustin** [comment]
I also found [DMI_HARDCODED_ABSOLUTE_FILENAME](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-code-contains-a-hard-coded-reference-to-an-absolute-pathname-dmi-hardcoded-absolute-filename) has this false negative, please read the following code example:
```java
public String path() {
  return ""d:\\test.txt"";
}
public void win32PathD() {
  new File(path());
}
```
SpotBugs should reported a warning at line 2 because it returns  hardcoded path, but no warnings. Hence, I think this is a false negative.
